pht_federated/aggregator/services/discovery/column_harmonization.py

```diff
@@ -0,0 +1,217 @@
+
+from fuzzywuzzy import fuzz
+
+from pht_federated.aggregator.schemas.dataset_statistics import DatasetStatistics
+
+
+def compare_two_datasets(
+    dataset_statistics: Tuple[DatasetStatistics, str],
+    aggregator_statistics: DatasetStatistics,
+) -> dict:
+    """
+    Compares two datasets in form of DatasetStatistics objects with respect to their column names
+    :param dataset_statistics: DatasetStatistics of local dataset
+    :param aggregator_statistics: DatasetStatistics of aggregator dataset
+    :return: Dictionary which lists the differences between the two datasets
+    """
+
+    dataset_name = dataset_statistics[1]
+    df_stats_dict = dataset_statistics[0].dict()
+    aggregator_stats_dict = aggregator_statistics.dict()
+
+    df_stats_dict = df_stats_dict["column_information"]
+    aggregator_stats_dict = aggregator_stats_dict["column_information"]
+
+    df_stats_dict_keys = [(x["title"], x["type"]) for x in df_stats_dict]
+    aggregator_stats_dict_keys = [
+        (x["title"], x["type"]) for x in aggregator_stats_dict
+    ]
+
+    # find intersection
+    (
+        intersection,
+        type_differences,
+        aggregator_keys_updated,
+        df_stats_dict_keys,
+    ) = intersection_two_lists(df_stats_dict_keys, aggregator_stats_dict_keys)
+
+    # find difference (Dataframe - Aggregator)
+    column_value_differences = list(
+        set(df_stats_dict_keys).difference(set(aggregator_keys_updated))
+    )
+    (
+        df_stats_dict_keys,
+        column_value_differences,
+        matched_column_names,
+    ) = fuzzy_matching_prob(
+        df_stats_dict_keys, aggregator_keys_updated, column_value_differences
+    )
+
+    # find difference (Aggregator - Dataframe)
+    column_value_differences2 = list(
+        set(aggregator_keys_updated).difference(set(df_stats_dict_keys))
+    )
+
+    difference_report = create_difference_report(
+        type_differences,
+        column_value_differences,
+        column_value_differences2,
+        matched_column_names,
+        dataset_name,
+    )
+
+    return difference_report
+
+
+def create_difference_report(
+    type_differences: List[List[Tuple[str, str]]],
+    column_value_differences: List[Tuple[str, str]],
+    column_value_differences2: List[Tuple[str, str]],
+    column_name_differences: List[List[Tuple[str, str]]],
+    dataset_name: str,
+) -> dict:
+    """
+    Transforms multiple types of mismatch errors between datasets into a summarized difference report
+    :param type_differences: List[List[Tuple[col_name_df, type], Tuple[col_name_aggregator, type]]] differences in type
+    :param column_value_differences: List[Tuple[col_name, type]] differences (Dataframe - Aggregator)
+    :param column_value_differences2: List[Tuple[col_name, type]] differences (Aggregator - Dataframe)
+    :param column_name_differences: List[List[Tuple[col_name_df, type], Tuple[col_name_aggregator, type]]] differences
+                                    in name only
```

Comment:
Duplicate type hints, When using type hints for arguments you don't need to specify them again in the docstrings